docs/research-software/mitgcm/mitgcm.md

```diff
@@ -99,3 +104,163 @@ each for up to one hour.
+
+Next, acquire your WebDAV credentials: <https://ecco.jpl.nasa.gov/drive> (second box from the top)
+
+Now, you can use wget to download the required forcing and input files:
+
+    wget -r --no-parent --user YOURUSERNAME --ask-password 
+      https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/input_forcing
+
+    wget -r --no-parent --user YOURUSERNAME --ask-password 
+      https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/input_init
+    
+    wget -r --no-parent --user YOURUSERNAME --ask-password
+      https://ecco.jpl.nasa.gov/drive/files/Version4/Release4/input_ecco
+
+### Compiling and running ECCOv4-r4
+
+Follow the instructions in the above section on building MITgcm on ARCHER2. These steps also work for ECCOv4-r4. You will need to create a build directory first:
+
+    cd MITgcm/ECCOV4/release4
+    mkdir build
+    cd build
+    
+Once you have compiled the model, you will have the mitgcmuv executable. 
+
+#### Issue with DOS formatting and end-of-namelist characters
+
+As of 16 December 2020, some of the ECCOv4-r4 files downloaded from the ECCO GitHub repository appear to be DOS formatted and/or have inconsistent end-of-namelist characters. This causes end-of-file runtime errors. The simplest solution appears to be running the dos2unix command in the namelist directory:
+
+    cd ../input_init/NAMELIST/
+    dos2unix data*
+  
+If you still get "end of file" runtime errors after running this command, you may need to manually replace the end-of-namelist "&" characters in some of the namelist files with "/". The files with "&" end-of-namelist characters are:
+
+    data.autodiff
+    data.salt_plume
+    data.layers
+    data.optim
+    data.ptracers
+    
+So far, this has only been tested with the gnu compiler. 
+
+#### Create run directory and link files
+
+In order to run the model, you need to create a run directory and link/copy the appropriate files. From the ``MITgcm/ECCOV4/release4`` directory:
+
+    mkdir run
+    cd run
+    
+    # link the data files
+    ln -s ../input_init/NAMELIST/* .
+    ln -s ../input_init/error_weight/ctrl_weight/* .
+    ln -s ../input_init/error_weight/data_error/* .
+    ln -s ../input_init/* .
+    ln -s ../input_init/tools/* .
+    ln -s ../input_ecco/*/* .
+    ln -s ../input_forcing/eccov4r4* .
+
+    python mkdir_subdir_diags.py
+    
+    # manually copy the mitgcmuv executable
+    cp -p ../build/mitgcmuv .
+
+For a short test run, edit the ``nTimeSteps`` variable in the file ``data``. Comment out the default value and uncomment the line reading ``nTimeSteps=8``. This is a useful test to make sure that the model can at least start up. 
+
+To run on ARCHER2, submit a batch script to the slurm scheduler. You can use the example slurm file from above, with the following modifications:
+
+    #SBATCH --job-name=ECCOv4r4-test
+    #SBATCH --time=1:0:0
+    #SBATCH --nodes=8
+    #SBATCH --tasks-per-node=12
+    #SBATCH --cpus-per-task=1
```

Comment:
If you are only using 12 MPI tasks per node, do you really want them confined to one NUMA region in this way or do you want them more evenly distributed across the node by increasing the `cpus-per-task` value?

Reply:
I'm using this setup based on @welucas2' Slurm script, although that was for a different setup. I'm happy to get some guidance here if you have a suggestion. What should I try increasing it to?

Reply:
I should note that we're not using OpenMP. We're only using MPI. Doesn't that mean it needs to be =1, or have I misunderstood something?

Reply:
@DanJonesOcean the `cpus-per-task` sets the stride between MPI processes which can be used to either make space for OpenMP threads or distribute MPI processes more evenly across the node. As this has been tested, I think it can just go in like this for now. You may want to see if distributing more evenly across sockets and NUMA regions improves things.